rs/embedders/src/wasm_utils/instrumentation.rs

```diff
@@ -173,8 +173,74 @@ impl InjectedImports {
     }
 }
 
-// Gets the cost of an instruction.
-pub fn instruction_to_cost(i: &Operator) -> u64 {
+pub fn instruction_to_cost(i: &Operator, mem_type: WasmMemoryType) -> u64 {
+    match mem_type {
+        WasmMemoryType::Wasm32 => instruction_to_cost_wasm32(i),
+        WasmMemoryType::Wasm64 => instruction_to_cost_wasm64(i),
+    }
+}
+
+// Gets the cost of an instruction in 64-bit mode.
+pub fn instruction_to_cost_wasm64(i: &Operator) -> u64 {
```

Comment:
Can we not inline this in the single `instruction_to_cost` function?

I'd imagine something like:
```
Operator::I32Load { .. }
        | Operator::I64Load { .. }
        | Operator::F32Load { .. }
        | Operator::F64Load { .. }
        | Operator::I32Load8S { .. }
        | Operator::I32Load8U { .. }
        | Operator::I32Load16S { .. }
        | Operator::I32Load16U { .. }
        | Operator::I64Load8S { .. }
        | Operator::I64Load8U { .. }
        | Operator::I64Load16S { .. }
        | Operator::I64Load16U { .. }
        | Operator::I64Load32S { .. }
        | Operator::I64Load32U { .. }
        | Operator::I32Store { .. }
        | Operator::I64Store { .. }
        | Operator::F32Store { .. }
        | Operator::F64Store { .. }
        | Operator::I32Store8 { .. }
        | Operator::I32Store16 { .. }
        | Operator::I64Store8 { .. }
        | Operator::I64Store16 { .. }
        | Operator::I64Store32 { .. } => match mem_type {
           WasmMemoryType::Wasm32 => 1,
           WasmMemoryType::Wasm64 => 2,
        },
```
The benefit I see is that we don't do a match on instructions twice for some instructions (it's a big enum, not sure how much of a performance hit there is but seems like a simple way). Also, we skip matching on wasm type for most instructions which is what we want.

Reply:
oh, I see, sure. I'll change it to something like this.